addons/mail/models/discuss/discuss_channel.py

```diff
@@ -476,8 +476,8 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                 member._bus_send("discuss.channel/joined", payload)
                 if post_joined_message:
                     notification = (
-                        _("joined the channel")
-                        if member.is_self
+                        _("%s joined the thread", member.partner_id.name) if member.is_self and self.parent_channel_id
+                        else _("%s joined the channel", member.partner_id.name) if member.is_self
                         else _("invited %s to the channel", member._get_html_link(for_persona=True))
```

Comment:
- prefer named placeholder: `%(user)s joined the channel`. This helps translators understand the context of sentence. Same remark for all translated string in the diff of this PR.
- The "invited" branching is also a partial sentence. I suppose this relies on dirty hack of adding author name in JS code. I would also fix it by putting the name of current user.

Reply:
There are many problems with notification: it has partial sentence poorly filled in JS, the notification language is the same for everyone (it's the person that made the add_members or manually joined the conversation), and so on.

I would say the dynamic change of name is the lowest importance of problem to fix. IMO I would put a higher priority in correct translation sentence, so we can remove the technical debt of dirty JS code and then improve dynamic part if we feel worth having it.

I don't have a strong preference of which way to start improving the code, but I would try to pick the one that reduces code maintenance. I'd say improving sentences to full and removing custom JS code (at the cost of this dynamic aspect) is an improvement.